csrc/ops/alias.cpp

```diff
@@ -629,6 +629,21 @@ TensorView* cat(
     return set(inputs.at(0));
   }
 
+  // short-circuit: If all inputs are already padded, assume correctness and cat
+  // them without padding. FusionTranslation adds the padOp for each tensor
+  // separately.
+  // TODO add correctness check against desired padding.
+  bool all_padded =
```

Comment:
> assume correctness and cat
  // them without padding. FusionTranslation adds the padOp for each tensor
  // separately.
  // TODO add correctness check against desired padding.

I think we needed to have the correctness checking here.
How are we going to identify cases where we explicitly pad a tensor before concat?

Reply:
I'm not sure how to check if two values are the same at the fusion definition without expression evaluator.

My only idea is to add a boolean flag for the translation, so this short circuit only applies then.

Reply:
I decided to detect if a `CatOp` follows a `PadOp` during translation. In this case, skip the `PadOp` so it is like the original script. Now there is no changes to `csrc/ops/alias.cpp`.

Reply:
sorry this falls off my radar. I'll take another look shortly